Merge Up Master Branch from Upstream

TaysirTayyab opened the pull request:
Pulls in latest changes from upstream on master.

micahhausler committed:
Add support for ClusterRoleBinding

pdecat committed:
Add EnvFrom to container schema as env_from

pdecat committed:
Add env_from to pod and replication_controller documention

pdecat committed:
Implement env_from mapping

pdecat committed:
Implement env_from reverse mapping

pdecat committed:
Fix bool pointers conversions

pdecat committed:
Add acceptance test

radeksimko committed:
test-infra: Allow testing against different K8S versions

radeksimko committed:
Merge pull request #89 from terraform-providers/t-expose-kube-version

test-infra: Allow testing against different K8S versions

apparentlymart committed:
Update README.md

jdwyah committed:
fix README to work with new github acct

hashicorp -> terraform-providers

tombuildsstuff committed:
Merge pull request #99 from jdwyah/patch-1

fix README to work with new github acct

nickschuch committed:
Update github.com/terraform-providers/terraform-provider-kubernetes to use k8s/client-go 1.9.0

Based on nickschuch/terraform-provider-kubernetes@aaff7e9.

nickschuch committed:
Update github.com/terraform-providers/terraform-provider-kubernetes to use k8s/client-go 1.9.0

Based on nickschuch/terraform-provider-kubernetes@aaff7e9.

sl1pm4t committed:
kubernetes_secret corrupts binary data on create

ofpiyush committed:
fix argument name in example

vbrednikov committed:
Fix link to secrets design proposal

sergei-ivanov committed:
Support `storage_class_name` in `kubernetes_persistent_volume`

pdecat committed:
Typo: config map => storage class

radeksimko committed:
Merge pull request #105 from ofpiyush/patch-1

fix argument name in example

radeksimko committed:
Merge pull request #107 from vbrednikov/master

Fix link to secrets design proposal

radeksimko committed:
Merge pull request #112 from pdecat/patch-1

Typo: config map => storage class

pdecat committed:
Add test case revealing issue #115 when adding labels to a resource that didn't have any

pdecat committed:
Add test case revealing issue #115 when adding annotations to a resource that didn't have any

pdecat committed:
Fix adding labels and annotations to a resource that didn't have any

pdecat committed:
Add test case revealing issue #115 when adding data to a config map resource that didn't have any

pdecat committed:
Add test case revealing issue #115 when adding data to a secret resource that didn't have any

pdecat committed:
Fix Flex persistent volume flattener/expander functions after upgrade of k8s libs to 1.9

pdecat committed:
Fix Flex persistent volume flattener/expander functions after upgrade of k8s libs to 1.9

radeksimko committed:
test-infra: Add more targets (kops + minikube)

radeksimko committed:
Merge pull request #103 from sl1pm4t/secret-binary-data

resource/kubernetes_secret: Prevent binary data corruption

radeksimko committed:
Update CHANGELOG.md

radeksimko committed:
Merge pull request #120 from terraform-providers/t-test-infra

test-infra: Add more targets (kops + minikube)

radeksimko committed:
test-infra: Expose google_zone from kops @ GCE

radeksimko committed:
Merge pull request #121 from terraform-providers/t-expose-google-zone

test-infra: Expose google_zone from kops @ GCE

radeksimko committed:
tests: Ignore internal k8s annotations+labels

radeksimko committed:
tests: Ignore resource_version in import tests

radeksimko committed:
Merge pull request #123 from terraform-providers/t-ignore-resource-version

tests: Ignore resource_version in import tests

radeksimko committed:
Merge pull request #122 from terraform-providers/t-ignore-internal-keys

tests: Ignore internal k8s annotations+labels

radeksimko committed:
tests: Fix wrong access_modes for PVC

radeksimko committed:
resource/kubernetes_persistent_volume_claim: Improve event log polling for warns

radeksimko committed:
tests: Fix PVC acceptance tests

radeksimko committed:
Add PV acc test with new field

radeksimko committed:
Merge pull request #124 from terraform-providers/t-gcp-pvc-access-mode

tests: Fix wrong access_modes for PVC

radeksimko committed:
Merge pull request #125 from terraform-providers/f-pvc-improve-event-polling

resource/kubernetes_persistent_volume_claim: Improve event log polling for warns

radeksimko committed:
Update CHANGELOG.md

radeksimko committed:
Merge pull request #111 from sergei-ivanov/pv_storage_class_name

resource/kubernetes_persistent_volume: Add support for storage_class_name

radeksimko committed:
Update CHANGELOG.md

radeksimko committed:
tests: Skip GCP-dependent test if not on GCP

radeksimko committed:
resource/kubernetes_service: Update external_ips correctly on K8S 1.8+

radeksimko committed:
tests: Add AWS EBS tests for PV + separate test for hostPath

radeksimko committed:
vendor: github.com/terraform-providers/terraform-provider-aws

radeksimko committed:
vendor: github.com/hashicorp/terraform/...@v0.11.3

radeksimko committed:
vendor: golang.org/x/net/idna@a04bdaca5b32abe1c069418fb7088ae607de5bd0

radeksimko committed:
vendor: github.com/hashicorp/go-getter/...@a686900cb3753aa644dc4812be91ceaf9fdd3b98

radeksimko committed:
Merge pull request #126 from terraform-providers/t-pv-skip

tests: Skip GCP-dependent test if not on GCP

radeksimko committed:
Merge pull request #127 from terraform-providers/b-svc-update-ext-ips

resource/kubernetes_service: Update external_ips correctly on K8S 1.8+

radeksimko committed:
Update CHANGELOG.md

radeksimko committed:
resource/kubernetes_persistent_volume: Mark persistent_volume_source as ForceNew on 1.9+

radeksimko committed:
resource/kubernetes_persistent_volume: Mark persistent_volume_source as ForceNew on 1.9+

radeksimko committed:
Merge pull request #128 from terraform-providers/vendor-hc-tf

vendor: github.com/hashicorp/terraform/...@v0.11.3

radeksimko committed:
vendor: github.com/aws/aws-sdk-go/...@v1.12.75

radeksimko committed:
Merge pull request #129 from terraform-providers/vendor-aws-provider

vendor: github.com/terraform-providers/terraform-provider-aws

radeksimko committed:
resource/kubernetes_limit_range: Avoid spurious diff when spec is empty

radeksimko committed:
Merge pull request #116 from pdecat/fix-adding-labels

Fix adding labels/annotations to resources or data to config maps/secrets when those were empty

radeksimko committed:
Update CHANGELOG.md

radeksimko committed:
resource/kubernetes_persistent_volume: Use correct operation when updating persistent_volume_source

radeksimko committed:
Merge pull request #130 from terraform-providers/t-aws-ebs

tests: Add AWS EBS tests for PV + separate test for hostPath

radeksimko committed:
Merge pull request #132 from terraform-providers/b-lr-empty-spec

resource/kubernetes_limit_range: Avoid spurious diff when spec is empty

radeksimko committed:
Update CHANGELOG.md

radeksimko committed:
Merge pull request #133 from terraform-providers/b-pv-correct-ops

resource/kubernetes_persistent_volume: Use correct operation when updating persistent_volume_source (1.8)

radeksimko committed:
Update CHANGELOG.md

radeksimko committed:
Fix failing test

radeksimko committed:
Fix failing test

radeksimko committed:
resource/kubernetes_pod: Bump timeout to 5 mins

radeksimko committed:
test-infra: Add DNS check to kops

radeksimko committed:
test-infra: Re-auth for kops@GCE deletion

radeksimko committed:
resource/*: Treat non-string label values as invalid

radeksimko committed:
Merge pull request #138 from terraform-providers/t-reauth-kops

test-infra: Re-auth for kops@GCE deletion

radeksimko committed:
Merge pull request #137 from terraform-providers/t-kops-dns-check

test-infra: Add DNS check to kops

radeksimko committed:
Merge pull request #136 from terraform-providers/b-pod-bump-timeout

resource/kubernetes_pod: Bump timeout to 5 mins

radeksimko committed:
Update CHANGELOG.md

radeksimko committed:
Merge pull request #135 from terraform-providers/b-label-vals-invalid

resource/*: Treat non-string label values as invalid

radeksimko committed:
Update CHANGELOG.md

radeksimko committed:
vendor: github.com/hashicorp/terraform/...@v0.11.4

radeksimko committed:
vendor: github.com/hashicorp/terraform/...@v0.11.4

radeksimko committed:
Merge pull request #148 from terraform-providers/vendor-tf-0.11.4

vendor: github.com/hashicorp/terraform/...@v0.11.4

radeksimko committed:
Merge pull request #139 from terraform-providers/b-pvs-forcenew-19

resource/kubernetes_persistent_volume: Mark persistent_volume_source as ForceNew on 1.9+

radeksimko committed:
Update CHANGELOG.md

radeksimko committed:
Update CHANGELOG.md

pdecat committed:
Refactor labelSelector schema and its flatteners/expanders functions for reuse (#140)

* Move labelSelector schema and its flatteners/expanders functions out of persistant volume claims files to their own files for reuse

* Simplify code

pdecat committed:
Refactor labelSelector schema and its flatteners/expanders functions for reuse (#140)

* Move labelSelector schema and its flatteners/expanders functions out of persistant volume claims files to their own files for reuse

* Simplify code

Commit:
v1.1.0

Commit:
v1.1.0

Commit:
Cleanup after v1.1.0 release

Commit:
Cleanup after v1.1.0 release

appilon committed:
add timeouts to pod resource

appilon committed:
add timeouts to pod resource

appilon committed:
configure timeouts for create and delete

appilon committed:
configure timeouts for create and delete

appilon committed:
change name validator to DNS1123 subdomain

appilon committed:
change name validator to DNS1123 subdomain

appilon committed:
acceptance test for new name validation

appilon committed:
acceptance test for new name validation

appilon committed:
Merge pull request #151 from appilon/91-k8s-pod-timeout

r/kubernetes_pod: Add timeout to pod resource

appilon committed:
Update CHANGELOG.md

appilon committed:
Update CHANGELOG.md

appilon committed:
simplify test config

appilon committed:
simplify test config

appilon committed:
Merge pull request #152 from appilon/87-k8s-secret-fix-validation

validators: change label validation to DNS1123 subdomain

appilon committed:
Update CHANGELOG.md

appilon committed:
Update CHANGELOG.md

appilon committed:
switch targetPort to string in schema
new acceptance test
update docs to reflect targetPort is optional

appilon committed:
switch targetPort to string in schema
new acceptance test
update docs to reflect targetPort is optional

appilon committed:
add forcenew on env var value field
add acc test

appilon committed:
add forcenew on env var value field
add acc test

appilon committed:
Merge pull request #154 from appilon/134-target_port-string-or-int

r/kubernetes_service: Switch targetPort to string

appilon committed:
Update CHANGELOG.md

appilon committed:
Update CHANGELOG.md

appilon committed:
Merge pull request #155 from appilon/98-container-env-force-new

r/kubernetes_pod: env var value change forces new pod

appilon committed:
Update CHANGELOG.md

appilon committed:
Update CHANGELOG.md

appilon committed:
add support for init containers
only allow container spec image to be updated by non-init containers
add acceptance checks for forced new
add acceptance test to Replication Controller

appilon committed:
add support for init containers
only allow container spec image to be updated by non-init containers
add acceptance checks for forced new
add acceptance test to Replication Controller

appilon committed:
switch target_port to string

appilon committed:
switch target_port to string

appilon committed:
Merge pull request #159 from appilon/fix-service-datasource

d/kubernetes_service: fix broken schema

appilon committed:
Update CHANGELOG.md

appilon committed:
Update CHANGELOG.md

radeksimko committed:
make: Add website + website-test targets

radeksimko committed:
make: Add website + website-test targets

radeksimko committed:
Merge pull request #160 from terraform-providers/f-make-website

make: Add website + website-test targets

appilon committed:
Merge pull request #156 from appilon/61-init-containers

r/kubernetes_pod: add support for init containers

appilon committed:
Update CHANGELOG.md

appilon committed:
Update CHANGELOG.md

katbyte committed:
README.md: correct provider organization & missing repository name

katbyte committed:
README.md: correct provider organization & missing repository name

Commit:
Update vendored libraries

Commit:
Update vendored libraries

paultyng committed:
Modify secret example per k8s docs

See https://kubernetes.io/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod

Fixes #79

paultyng committed:
Modify secret example per k8s docs

See https://kubernetes.io/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod

Fixes #79

paultyng committed:
Merge pull request #165 from terraform-providers/doc-fix

Modify secret example per k8s docs

paultyng committed:
Update CHANGELOG.md

paultyng committed:
Update CHANGELOG.md

antonbabenko committed:
Updated example of providers block

antonbabenko committed:
Updated example of providers block

radeksimko committed:
Merge pull request #170 from antonbabenko/updated-example

Updated example of providers block

radeksimko committed:
Update README.md

radeksimko committed:
Update README.md

mrolla committed:
Fix basic auth example in docs

mrolla committed:
Fix basic auth example in docs

radeksimko committed:
Merge pull request #179 from mrolla/fix-documentation

Fix basic auth example in docs

coryflucas committed:
Fix link to storage class parameters

coryflucas committed:
Fix link to storage class parameters

radeksimko committed:
Merge pull request #180 from coryflucas/patch-1

Fix link to storage class parameters

Commit:
v1.2.0

Commit:
v1.2.0

Commit:
Cleanup after v1.2.0 release

Commit:
Cleanup after v1.2.0 release

alexsomesan committed:
Minikube on KVM at Packet

alexsomesan committed:
Minikube on KVM at Packet

alexsomesan committed:
Auto configure cluster node CPU and RAM size based on host sizes

alexsomesan committed:
Auto configure cluster node CPU and RAM size based on host sizes

alexsomesan committed:
Document minikube settings

alexsomesan committed:
Document minikube settings

alexsomesan committed:
Merge pull request #187 from terraform-providers/fix-packet-infra

KVM minikube cluster on Packet

alexsomesan committed:
Merge pull request #162 from cosmopetrich/kube-client-1.10

Upgrade k8s client libs to 1.10

nickschuch committed:
New resource: Deployment

alexsomesan committed:
Adapt package import paths for newer K8S client library

alexsomesan committed:
Merge pull request #101 from nickschuch/deployment

Deployment

alexsomesan committed:
Fix patch operations on resource update + other small fixes

alexsomesan committed:
Tests: ServiceAccount subjects don't belong to the default RBAC API group.

alexsomesan committed:
Merge pull request #73 from micahhausler/f-k8s-clusterrolebinding

Add support for ClusterRoleBinding

alexsomesan committed:
Merge pull request #82 from pdecat/f-container-env_from

Add support for envFrom on container definitions

alexsomesan committed:
Fix Deployment attributes expansion (#191)

* Fix Deployment attributes expansion

* Bubble up errors from expandPodTemplate

* Refresh deployment on metadata returned by Create call

* Deployment acceptance tests

* Replace Create timer-based waiter with a resource.RetryFunc

paultyng committed:
Update CHANGELOG.md

Commit:
v1.3.0

Commit:
Cleanup after v1.3.0 release

alexsomesan committed:
Add documentation for kubernetes_deployment resource (#194)

* Add documentation for kubernetes_deployment resource

* Docs - convert verbatim URLs to hyperlinks.

* Fix broken links

alexsomesan committed:
Bump Go version to 1.10 on Travis CI. (#196)

* Bump Go version to 1.10 on Travis CI.

* Try exact Go version

* Quotes around Go version

pdecat committed:
Remove maintainer section with broken link (#197)

Linked to #178

olib963 committed:
Allow automount service account token on the service account resource (#198)

pdecat committed:
Fix deployment selector description (#208)

Description taken from https://github.com/terraform-providers/terraform-provider-kubernetes/blame/13b7e681cd620f8344395aa8fdbc29b4af590cfe/website/docs/r/deployment.html.markdown#L99

pdecat committed:
Fix deployment selector documentation (#209)

This is actually *Optional*: https://github.com/terraform-providers/terraform-provider-kubernetes/blob/6d4babb71bd11c58542970a69d67f0a82e52c5aa/kubernetes/resource_kubernetes_deployment.go#L81

andyvanosdale committed:
Wait for deployment rollout status (#211)

* Wait for deployment rollout status

replicationcontrollers are not created for a deployment. So we need to
wait for the deployment itself to finish.

juliendangers committed:
Add ReclaimPolicy attribute to Storage class (#213)

* feat(StorageClass): add ReclaimPolicy attribute

* doc(StorageClass): add new attribute reclaim_policy

* chore(format): reindent resource_kubernetes_storage_class.go

alexsomesan committed:
Fix example configuration for ServiceAccount token automount test. (#222)

alexsomesan committed:
Kops on AWS test env - create state S3 bucket via Terraform (#223)

* Kops on AWS test env - create state S3 bucket via Terraform

alexsomesan committed:
Fix deployment acc tests (#228)

* Constantize test resource name

* Fix detection of K8S config context & be aware of KUBECONFIG env var

* Scale down deployment tests to fit GKE tighter resource constraints

* Pin down cluster node instance type

alexsomesan committed:
New resource: StatefulSets - CRUD & acceptance tests (#203)

* StatefulSets - Create/Read/Update/Destroy
* StatefulSet acceptance tests
* Documentation for StatefulSet

alexsomesan committed:
Complete run of terrafmt.rb - format inline HCL in ACC tests (#230)

alexsomesan committed:
Bring back 'wait_until_bound' parameter on PVCs (#232)

alexsomesan committed:
PVCs: expose 'wait_until_bound' only on stand-alone PVCs, not embedded ones (#234)

alexsomesan committed:
Update CHANGELOG.md

Commit:
v1.4.0

Commit:
Cleanup after v1.4.0 release

alexsomesan committed:
Convert returned types to pointers (#231)

Convert returned types to pointers

pdecat committed:
Fix Deployment acceptance tests (#248)

alexsomesan committed:
Fix formatting issues

alexsomesan committed:
Revert "Fix formatting issues"

This reverts commit 7acf8d56631895e245aba829e7751ba2145cccbb.

olib963 committed:
Add argument doc (#253)

pdecat committed:
Add allow_privilege_escalation to container security contexts attributes (#249)

Add allow_privilege_escalation property to container security context, update documentation and acceptance tests

pdecat committed:
Fix updates of stateful set images (#252)

* Add acceptance test to demonstrate issue with stateful set image updates

* Fix stateful set template spec updates

huikaihoo committed:
Fix the id of storage class when update (#239)

Stelminator committed:
add kubernetes_secret datasource (#243)

adds data_source_kubernetes_secret

https://github.com/terraform-providers/terraform-provider-kubernetes/issues/241

alexsomesan committed:
GKE test infra writes kubeconfig file (#259)

* GKE test infra writes kubeconfig file

tete17 committed:
Typos fix (#264)

* Typos fix

meyskens committed:
Add support for volumeBindingMode in storageClass (#266)

* Add support for volumeBindingMode in storageClass

pdecat committed:
Update changelog for #249 and #252 (#257)

* Update changelog for #249

* Update changelog for #252

wjam committed:
Add kubernetes_role resource (#226)

Add kubernetes_role resource - a new resource to allow the management of `role` resources in Terraform.

hardchiu committed:
Add NodeAffinity config to persistence volume (#219)

Added node-affinity support for persistence volume.
Added waiting for destroying persistence volume, which kubernetes test do similar things on destroy:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#WaitForPersistentVolumeDeleted

meyskens committed:
Add support for local PVs (#267)

Add support for local PVs

alexsomesan committed:
Upgrade to Go 1.11 + modules (#269)

* Remove old vendor

* Go 1.11 formatting

* Initial go.mod

* go mod tidy

* More explicit PVC types to make Go 1.11 happy

* Re-vendor from go modules

* provider: Require Go 1.11 in TravisCI and README
provider: Run go fix
provider: Run go fmt

pdecat committed:
Add network policy resource with support for 1.8+ fields (#118)

Add kubernetes_network_policy resource

pdecat committed:
Add  pod metadata to replication controller spec template (#193)

* Add pod metadata to replication controller spec template
* Mark affected attributes as deprecated to allow for a migration path

jhoblitt committed:
trivial .travis.yml cleanups (#271)

* rm sudo key

sudo is now always enabled:

https://blog.travis-ci.com/2018-11-19-required-linux-infrastructure-migration

* make yamllint 1.13.0 happy

wjam committed:
Add kubernetes_role_binding resource (#235)

Add a new resource "kubernetes_role_binding"

alexsomesan committed:
Update CHANGELOG summary for 1.5.0

Commit:
v1.5.0

Commit:
Cleanup after v1.5.0 release

pdecat committed:
Dev: fix test infra for GKE (#277)

* Fix typo in GKE test-infra README and add kubeconfig to .gitignore to avoid committing credentials by mistake

* Update README

strixBE committed:
typo (#279)

s/grand/grant/

MarvelOrange committed:
Move `automount_service_account_token` (#278)

The `automount_service_account_token` should not be in the metadata block.

chrisleck committed:
Add DaemonSet and ClusterRole resources (#229)

Add DaemonSet and ClusterRole resources

alexsomesan committed:
Remove redundant assignments (#282)

TaysirTayyab commented:
Requesting admin override on the failing test, since this is a direct pull from upstream.

sl1pm4t commented:
Hey. Just FYI I’m not going to be able to look at this properly for another few days since I’m on vacation with pathetic internet connection.

wjam committed:
Alter ClusterRole validation to closer match that of Kubernetes (#288)

This fix makes it possible to have rules which are `nonResourceUrls`.
Partially fixes #285

pablo-ruth committed:
select default_secret for service account if there is more than one (#281)

Select default_secret for service_account if there is more than one

Co-Authored-By: pablo-ruth <contact@pablo-ruth.fr>

alexsomesan committed:
Add kubernetes_secret datasource to docs sidebar. (#298)

alexsomesan committed:
Add test infrastructure for AKS and EKS (#291)

* Add test environment infrastructure for Azure AKS
* Add test environment infrastructure for Amazon EKS

alexsomesan committed:
Fix github cookie for go-getting in Travis CI (#301)

* Fix github cookie for go-getting in CI

* Remove vendor-status make target

pdecat committed:
Remove -i from 'go test' to allow running unit tests locally with go 1.11.5 and GO111MODULE=on (#300)

pdecat committed:
Add publish_not_ready_addresses to kubernetes_service (#306)

* Add `publish_not_ready_addresses` to `kubernetes_service`

pdecat committed:
Set min_master_version to kubernetes_version to match node_version (#307)

rremer committed:
Declare 'env' argument type correctly in pod config (#304)

pdecat committed:
Fix service datasource after #306 broke it (#313)

alexsomesan committed:
Update CHANGELOG.md

Commit:
v1.5.1

Commit:
Cleanup after v1.5.1 release

alexsomesan committed:
GKE test infra: base template path on module location (#330)

Commit:
Add nil check for configmap volume mode (#332)

alexsomesan committed:
EKS test infra: wait for API to be ready (#333)

alexsomesan committed:
Fix `api_group` attribute attribute of RBAC subjects. (#331)

* Fix RBAC subject schema

* Test all ClusterRoleBinding subject types

* Test all RoleBinding subject types

* Drop the default value on api_group in role_ref.

alexsomesan committed:
Bump EKS module versio to fix CI (#336)

alexsomesan committed:
Re-vendor after go mod tidy (#337)

alexsomesan committed:
Update CHANGELOG.md

Commit:
v1.5.2

Commit:
Cleanup after v1.5.2 release

TaysirTayyab commented:
@sl1pm4t no worries, I wanted to merge up to master to get the RBAC resources from upstream to get update/import  functionality for them in this build.

alexsomesan committed:
Make worker count configurable in GKE test infra (#354)

* Make worker count configurable in GKE test infra

* Fall back to 3 nodes by default (stay in quota).

mshannon1123 committed:
Add endpoint resource (#167) (#350)

* Add endpoint resource (#167)

pablo-ruth committed:
continue instead of break during secrets parsing + add retry error when token secret is not already available (#349)
